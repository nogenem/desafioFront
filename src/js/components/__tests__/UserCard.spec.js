import UserCard from "../UserCard";

const testData = {
  name: "test",
  email: "test@test.com",
  bio: "some bio",
  avatar_url: "http://site.com/image.png",
  followers: "0",
  following: "0"
};

it("renders correctly when passed all props", () => {
  const output = UserCard.render(testData);
  expect(output).toMatchSnapshot();
});

it("renders correctly when not passed all props", () => {
  const output = UserCard.render({
    ...testData,
    email: undefined,
    bio: undefined,
    avatar_url: undefined
  });
  expect(output).toMatchSnapshot();
});
