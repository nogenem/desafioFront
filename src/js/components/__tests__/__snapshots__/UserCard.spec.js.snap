// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when not passed all props 1`] = `
"
  <div id=\\"userCard\\" class=\\"row align-items-center justify-content-center\\">
    <div class=\\"card col-6\\">
      <div class=\\"card-body\\">
        <div class=\\"row\\">
          <div class=\\"col-12 col-lg-8 col-md-6\\">
            <h3 class=\\"mb-0 text-truncated\\">test</h3>
            <p class=\\"lead\\"></p>
            <p>
              
            </p>
          </div>
          <div class=\\"col-12 col-lg-4 col-md-6 text-center\\">
            <img src=\\"\\" alt=\\"\\" class=\\"mx-auto rounded-circle img-fluid\\">
          </div>
          </br>
          <div class=\\"col-12 col-lg-6 text-center\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Seguidores</small>
          </div>
          <div class=\\"col-12 col-lg-6 text-center\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Seguidos</small>
          </div>
        </div>
      </div>
    </div>
  </div>
"
`;

exports[`renders correctly when passed all props 1`] = `
"
  <div id=\\"userCard\\" class=\\"row align-items-center justify-content-center\\">
    <div class=\\"card col-6\\">
      <div class=\\"card-body\\">
        <div class=\\"row\\">
          <div class=\\"col-12 col-lg-8 col-md-6\\">
            <h3 class=\\"mb-0 text-truncated\\">test</h3>
            <p class=\\"lead\\">test@test.com</p>
            <p>
              some bio
            </p>
          </div>
          <div class=\\"col-12 col-lg-4 col-md-6 text-center\\">
            <img src=\\"http://site.com/image.png\\" alt=\\"\\" class=\\"mx-auto rounded-circle img-fluid\\">
          </div>
          </br>
          <div class=\\"col-12 col-lg-6 text-center\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Seguidores</small>
          </div>
          <div class=\\"col-12 col-lg-6 text-center\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Seguidos</small>
          </div>
        </div>
      </div>
    </div>
  </div>
"
`;
